File: Cpp/1748_Sum_of_Unique_Elements.cpp
class Solution {
public:
    int sumOfUnique(vector<int>& nums) {
        int result = 0;
        unordered_map<int, int> frequency;
        for(auto& num : nums){
            if(frequency.find(num) != frequency.end()){
                result -= frequency[num] * num;
                frequency[num] = 0;
                continue;
            }
            result += num;
            ++frequency[num];
        }        
        
        return result;
    }
};

File: Kotlin/1748_Sum_of_Unique_Elements.kt
class Solution {
    fun sumOfUnique(nums: IntArray): Int {
        var result = 0
        val frequency = HashMap<Int, Int>()
        
        for(num in nums) {
            if(frequency.contains(num)) {
                result -= frequency.getOrDefault(num, 0) * num
                frequency[num] = 0
                continue
            }
            result += num
            frequency.put(num, frequency.getOrDefault(num, 0) + 1)
        }

        return result
    }
}